### DISCUSSION TITLE
Attempting to handle reconnections myself throws "socketio.exceptions.ConnectionError: Already connected"

### DISCUSSION THREAD

**Discussion by devkrxspl** [miguelgrinberg/python-socketio#773] [Q&A] [answered]
When trying to handle reconnections myself, it always throws a `socketio.exceptions.ConnectionError: Already connected` error when doing `sio.connect("url")` in the disconnect handler.

**Code:**
```py
@sio.event
def disconnect():

    print("Connection lost.");
    
    try:
        sio.connect('url');
    except:
        print("Reconnection attempt failed.");

        while (True):
            try:    
                sio.connect('url');
                break;
            except Exception as e:
                print(e)
            
            print("Attempting to reconnect...");
            time.sleep(1);
```
(I've also already set `reconnections=False`). In the disconnect event, `sio.connected` also seems to be true. I've also tried doing `sio.disconnect()` in the while loop, doesn't seem to do anything.

**Comment by miguelgrinberg** [comment, marked as answer]
You can't issue a connection from an even handler. You can use a for-loop in the top level of your script, combined with the `sio.wait()` method. Something like this:

```python
while True:
    try:
        sio.connect(url)
    except Exception as e:
        print(e)
    else:
        sio.wait()

    print("Attempting to reconnect...");
    time.sleep(1)
```
